sopel/bot.py

```diff
@@ -594,27 +594,26 @@ def rate_limit_info(
     ) -> tuple[bool, Optional[str]]:
         if trigger.admin or rule.is_unblockable():
             return False, None
-
+        nick = trigger.nick
         is_channel = trigger.sender and not trigger.sender.is_nick()
         channel = trigger.sender if is_channel else None
 
-        at_time = trigger.time
-
-        user_metrics = rule.get_user_metrics(trigger.nick)
+        user_metrics = rule.get_user_metrics(nick)
         channel_metrics = rule.get_channel_metrics(channel)
         global_metrics = rule.get_global_metrics()
```

Comment:
With `mypy` 1.12.0 (released today), `channel_metrics = rule.get_channel_metrics(channel)` causes a type-check error at R602 in this patch's version of the file:

```
sopel/bot.py:604: error: Argument 1 to "get_channel_metrics" of "AbstractRule" has incompatible type "Any | None"; expected "Identifier"  [arg-type]
```

Since the revised code doesn't use the `RuleMetrics` objects in this section any more, the block of `*_metrics` assignments here can be removed.

That won't fix the other problem, though, which is that both `rule.is_channel_rate_limited()` and `rule.get_channel_metrics()` expect `Identifier` and do not accept `None`.

Reply:
And that is fixed as well.